### PR TITLE
Command cleanup

### PR DESCRIPTION
**Pull request by darrenmce**
this is based on the branch of PR #4 (so you can ignore the first two commits)

see commit log for details

### CODE REVIEW

File: tt_completion [hunk 1]
```diff
@@ -38,41 +41,41 @@ tt_complete() {
 		case "$cmd" in
 		tt|"")
 			completion_base $word ;;
-		manage)
-			completion_manage $word ;;
-		create)
+		create|browse)
 			completion_all_services $word ;;
-		destroy|recreate|ip|browse)
+		destroy|recreate|ip)
```
Comment (mpareja) [inline tt_completion]: Good catch on `browse` deserving completion for non-running services.

File: tt_completion [hunk 2]
```diff
@@ -1,8 +1,11 @@
 #!/usr/bin/env bash
 
-COMMANDS="api browse check clone create destroy kue install ip logs manage pull rabbit recreate restart upgrade usage"
-DATA_COMMANDS="aray ongoing"
-MANAGE_COMMANDS="build pull push usage"
+BASE_COMMANDS="browse check clone create dc destroy install ip pull recreate restart upgrade usage workspace"
+WORKSPACE_COMMANDS="init add rm ls use current upgrade"
+
+tarantino() {
```
Comment (mpareja) [inline tt_completion]: Any particular reason for this indirection?

File: tt_completion [hunk 3]
```diff
@@ -38,41 +41,41 @@ tt_complete() {
...
-		create)
+		create|browse)
 			completion_all_services $word ;;
-		destroy|recreate|ip|browse)
+		destroy|recreate|ip)
 			completion_running_services $word ;;
 		pull|clone|check)
 			completion_services $word ;;
+		workspace)
+			completion_workspace $word ;;
```
Comment (mpareja) [inline tt_completion]: I see `$word` passed in but I don't see it used. Is this parameter necessary?

File: workspace.sh [hunk 4]
```diff
@@ -16,7 +16,7 @@ ws_usage() {
 	echo '  ls - list registered workspaces '
 	echo '  use <workspace> - switch to a different workspace'
 	echo '  current - display the current workspace and directory'
-	echo '  upgrade [workspace] - fetch latest workspace definition, update docker images'
+	echo '  upgrade [workspace] - NOT YET IMPLEMENTED: fetch latest workspace definition, update docker images'
```
Comment (mpareja) [inline workspace.sh]: Looking forward to this!

File: tt_completion (hunk 3 above)
Comment (darrenmce) [inline tt_completion]: just keeping the signature consistent, the implementation of the function might make use of it

**Comment by darrenmce** [comment]
added some more cleanup commits

**Comment by mpareja** [comment]
Okay, I'm going to merge this. Going to use GitHub's rebase and merge feature since this was built on top of the previously merged branch!

**Pull request merged**